salon_creator/lib/authentication/sign_in.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+
+import 'package:salon_creator/models/user_model.dart';
+import 'package:salon_creator/constant/constants.dart' as constants;
+
+UserModel currentSignedInUser = UserModel();
+final db = FirebaseFirestore.instance;
+
+Future<UserCredential> signInWithGoogle() async {
+  final GoogleSignInAccount googleUser = await GoogleSignIn().signIn();
+  final GoogleSignInAuthentication googleAuth = await googleUser.authentication;
+  final credential = GoogleAuthProvider.credential(
+    accessToken: googleAuth.accessToken,
+    idToken: googleAuth.idToken,
+  );
+
+  return await FirebaseAuth.instance.signInWithCredential(credential);
+}
+
+Future addUserToDatabase() async {
+  final _auth = FirebaseAuth.instance;
+  final User user = _auth.currentUser;
```

Comment:
```suggestion
  final User user = FirebaseAuth.instance.currentUser;
```
variable _auth is redundant